chef-helpers/lib/chef-helpers/platform_family.rb

```diff
@@ -0,0 +1,170 @@
+#
+# Copyright:: Copyright 2018-2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef-helpers/internal"
+
+module ChefHelpers
+  module PlatformFamily
+
+    extend self
+
+    #
+    # NOTE CAREFULLY: Most node['platform_family'] values should not appear in this file at all.
+    #
+    # For cases where node['os'] == node['platform_family'] == node['platform'] then
+    # only the platform helper should be added.
+    #
+    # For cases where there are more than one platform in the platform family, but the platform_family
+    # name duplicates one of the platform names, then it should be added here with the _family? suffix.
+    # (e.g. fedora_platform?, debian_platform?, etc).
+    #
+
+    # Determine if the current node is a member of the debian family.
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def debian_platform?(node)
+      node["platform_family"] == "debian"
+    end
+    # NOTE: debian? matches only the exact platform
+
+    # Determine if the current node is a member of the fedora family.
+    #
+    # @param [Chef::Node] node
+    #
+    # @return [Boolean]
+    #
+    def fedora_platform?(node)
+      node["platform_family"] == "fedora"
+    end
```

Comment:
note:  why is this `_platform?` and not `_family?` and what does chef-sugar do and why are we doing this?